sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1810,6 +1818,9 @@ pi_result piContextRelease(pi_context Context) {
 
   assert(Context);
   if (--(Context->RefCount) == 0) {
+    // Clean up any live memory associated with Context
+    Context->finalize();
```

Comment:
let's move all the context de-initialization code to the "finalize" then

Reply:
Done.